crates/services/p2p/src/service.rs

```diff
@@ -475,29 +486,32 @@ where
                         let peer = self.p2p_service.get_peer_id_with_height(&height);
                         let _ = self.p2p_service.send_request_msg(peer, request_msg, channel_item);
                     }
-                    Some(TaskRequest::GetSealedHeaders { block_height_range, channel: response}) => {
-                        let request_msg = RequestMessage::SealedHeaders(block_height_range.clone());
+                    Some(TaskRequest::GetSealedHeaders { block_height_range, from_peer, channel: response}) => {
+                        let request_msg = RequestMessage::SealedHeaders(block_height_range);
                         let channel_item = ResponseChannelItem::SealedHeaders(response);
-
-                        // Note: this range has already been check for
-                        // validity in `SharedState::get_sealed_block_headers`.
-                        let block_height = BlockHeight::from(block_height_range.end - 1);
-                        let peer = self.p2p_service
```

Comment:
Yes, I agree. I will investigate how to maintain this behaviour. 

The challenge with the existing approach was that the interface doesn't guarantee that the headers in a streamed batch were from the same peer when requesting transactions. Each header was attached to a  `SourcePeer`. This means we couldn't make a single request for batch transactions because we couldn't guarantee that all the peers were the same at the service level. 

This was an experiment to see if we could work around that by ensuring that all headers come from the same peer - this way, we can be sure that each request for batch transactions is guaranteed to fulfil. 

I will investigate to understand if there is any guarantee of a 1-1 match of a batch of headers to a batch of transactions at the P2P level - if so, we can reorganize the service level to reflect that guarantee and maintain random peer selection per batch.